### PR TITLE
Challenge/topk kmoe

### PR DESCRIPTION
**Pull request by emharsha1812**
### Description
This PR introduces a new medium challenge: MoE Top-K Gating (Mixture of Experts Gating).

### Challenge Details
*   **Goal**: Implement the Top-K Gating mechanism used in MoE models. For a given logit matrix `(M, E)`, identify the top `k` experts per token, extract their indices, and compute their softmax-normalized mixing weights.
*   **Key Concepts**: Parallel reduction, parallel top-k selection, softmax, MoE architecture.

### Changes
*   Added `challenges/medium/66_moe_topk_gating`
*   Included `challenge.html` (problem statement + math formulation).
*   Added `challenge.py` (reference implementation using `torch.topk` + `softmax`).
*   Added starter templates for CUDA, PyTorch, Triton, Mojo, JAX, and CuTe.

### Verification
*   Reference implementation verified with custom test script.
*   Starter code generated and cleaned to match repository standards.

### CODE REVIEW

**Comment by kunal-mansukhani** [comment]
Nothing to review @emharsha1812

**Comment by kunal-mansukhani** [comment]
@claude review this PR

**Comment by kunal-mansukhani** [comment]
> **Claude encountered an error** —— [View job](https://github.com/AlphaGPU/leetgpu-challenges/actions/runs/21620350542)
> 
> ```
> Command failed: git fetch origin --depth=20 challenge/topk-kmoe
> ```
> 
> I'll analyze this and get back to you.

@claude use `gh` cli to checkout the branch

**Comment by kunal-mansukhani** [comment]
@claude Review this PR

**Comment by kunal-mansukhani** [comment]
@claude review this PR
